Add automated daily updates for M3 data

eemeli opened the pull request:
See #22 
CC @nordzilla, @zbraniecki 

This adds a minimally functional `GitSource` as an alternative to `HgSource` (previously just `Source`) and a corresponding CLI option `--git`, which lets the script work with git repositories as well as Mercurial ones. For now, this functionality only extends to the features used by `--use-current-revision`.

Using that, a scheduled (daily at 6AM UTC) GitHub Actions task is added which checks out the latest commit (i.e. "central") from the [mozilla/gecko-dev](https://github.com/mozilla/gecko-dev) mirror and generates the M3 data for it, and then updates the `gh-pages` branch.

To effect these changes and get a decent grasp of what the script is really doing, I've added some type hints which effectively require Python 3.9, and a pip `requirements.txt` file.

Before merging this, we should update the last few weeks' data to the `gh-pages` branch, as this script will only run on the latest commit.

To test the action, I've been manually running it on my fork of this repo: https://github.com/eemeli/arewefluentyet.com/actions/workflows/update.yaml

flodolo commented on requirements.txt:
```diff
@@ -0,0 +1 @@
+compare-locales>=8.2
```
I would put this file within the `.github` folder, since it's relevant for automation.

eemeli replied:
Tbh it's relevant in any environment where the code is run. A `Pipfile` would also be an alternative.

flodolo commented on .github/workflows/update.yaml:
```diff
@@ -0,0 +1,42 @@
+      - run: pip install -r requirements.txt
+
+      - name: Generate new data
+        run: >
```
TIL about `>` 👍🏼

eemeli replied:
It's one of the five YAML scalar formats. Because of course we need five. In this case, it's useful because it lets the single-line command be split on multiple lines.

flodolo commented on src/arewefluentyet/source.py:
```diff
@@ -8,10 +8,60 @@ def parse_date(input):
 
 class Source:
-    def __init__(self, path):
+    current_revision: str | None = None
```
I'm not familiar with the syntax, but why the assignment to None if it's already part of the constructor method? Or am I misunderstanding this code completely?

eemeli replied:
```suggestion
    current_revision: str | None
```

zbraniecki commented:
One concern I have is that I didn't make the front-end ready for such volume of data. We haven't updated the use of Chart.js since inception of AWFY and I don't know how it'll handle 350+ datapoints per year (vs 50).

It may just work, or it may start looking bad, and I'd suggest verifying before landing this :)

eemeli commented:
> @zbraniecki:
> One concern I have is that I didn't make the front-end ready for such volume of data. We haven't updated the use of Chart.js since inception of AWFY and I don't know how it'll handle 350+ datapoints per year (vs 50).
> 
> It may just work, or it may start looking bad, and I'd suggest verifying before landing this :)

Good point. I'm working on the front-end changes next, will verify that before landing this.

flodolo commented:
> Good point. I'm working on the front-end changes next, will verify that before landing this.

For what's worth, I have graphs with daily data over almost 3 years ([example](https://test.flod.org/firefox-l10n/?timeframe=all&locale=fi)), and chart.js seems to be coping well.

eemeli commented:
I [updated](https://github.com/projectfluent/arewefluentyet.com/compare/8ac95f4202b7a77c50de418327664f4971e61150...b8a130b2358f30f50455adca6e6d67a17f0a2de4) the `gh-pages` branch so that it no longer provides UI for switching away from the default M3 milestone view.

While at that, I also tested that it should now work with daily rather than weekly updates. This did require a small fix to `getBarPosition()`, which was occasionally failing a bit oddly around the end of the graph. I'll need to track how the UI performs once we start getting in real data, as there may be some glitches with the display of the hover legend.

I also updated the JS/CSS/HTML code style to Prettier, and fixed a few other bugs that I noticed.
